dev/DynamicDependency/MsixDynamicDependency.cpp

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "msixdynamicdependency.h"
+
+#include "MddDetourPackageGraph.h"
+#include "PackageDependencyManager.h"
+#include "PackageGraphManager.h"
+
+namespace MddCore
+{
+bool IsStaticPackageGraphEmpty()
+{
+    // Check the static package graph
+    UINT32 n = 0;
+    const auto rc{ GetCurrentStaticPackageInfo(PACKAGE_FILTER_HEAD, &n, nullptr, nullptr) };
+    (void) LOG_HR_IF_MSG(HRESULT_FROM_WIN32(rc), (rc != APPMODEL_ERROR_NO_PACKAGE) && (rc != ERROR_INSUFFICIENT_BUFFER), "GetCurrentStaticPackageInfo rc=%d", rc);
+    return rc == APPMODEL_ERROR_NO_PACKAGE;
+}
+}
+
+STDAPI MddTryCreatePackageDependency(
+    PSID user,
+    _In_ PCWSTR packageFamilyName,
+    PACKAGE_VERSION minVersion,
+    MddPackageDependencyProcessorArchitectures packageDependencyProcessorArchitectures,
+    MddPackageDependencyLifetimeKind lifetimeKind,
+    PCWSTR lifetimeArtifact,
+    MddCreatePackageDependencyOptions options,
+    _Outptr_result_maybenull_ PWSTR* packageDependencyId) noexcept try
+{
+    *packageDependencyId = nullptr;
+
+    // Dynamic Dependencies requires a non-packaged process
+    RETURN_HR_IF(E_NOTIMPL, !MddCore::IsStaticPackageGraphEmpty());
```

Comment:
nit: E_INVALID or similar?  not impl sounds like we will do this in the future maybe

Reply:
E_NOTIMPL = "Not implemented"
ERROR_NOT_SUPPORTED = "The request is not supported."
Different values and messages, but in practice in COM-world E_NOTIMPL means the same as ERROR_NOT_SUPPORTED

Reply:
I think I represent the "COM world" and I very much do not agree.  Not blocking but we should clean this up on the next pass.  ERROR_NOT_SUPPORTED would be (a lot) better.

Reply:
I'll buy that for a dollar :-)